int16 object_id